integrade/tests/aws_utils.py

```diff
@@ -0,0 +1,164 @@
+
+import boto3
+
+from integrade import config
+
+
+def wait_until_running(profile_and_id):
+    """Wait until an instance is running.
+
+    :params: tuple of (aws_profile_name, instance_id)
+
+    Note: input is taken in as a tuple to facilitate calling this with
+        ``multiprocessing.pool.Pool.map``.
+    """
+    (aws_profile, ec2_instance_id) = profile_and_id
+    session = aws_session(aws_profile)
+    instance = session.resource('ec2').Instance(ec2_instance_id)
+    instance.wait_until_running()
+
+
+def terminate_instance(profile_and_id):
+    """Terminate an instance and wait until it is terminated.
+
+    :params: tuple of (aws_profile_name, instance_id)
+
+    Note: input is taken in as a tuple to facilitate calling this with
+        ``multiprocessing.pool.Pool.map``.
+    """
+    (aws_profile, ec2_instance_id) = profile_and_id
+    session = aws_session(aws_profile)
+    instance = session.resource('ec2').Instance(ec2_instance_id)
+    instance.terminate()
+    instance.wait_until_terminated()
+
+
+def create_instances(aws_profile, image_name, count=1):
+    """Create instances from a named image.
+
+    :param aws_profile: (string) Name of profile as defined in config file
+    :param image_name: (string) Name of image as defined in config file
+    :param count: (int) Number of instances to create.
+
+    :returns: (json) API response from AWS.
+
+    The instances will be created using the named AMI as found in the
+    config file for the given profile.
+
+    The profile must be named and have its credentials stored in ~/.aws/config
+    The images are associated with the profiles in
+    ~/.config/integrade/aws_config in the following manner:
+
+    aws:
+      profiles:
+        name_of_profile1:
+          images:
+            name_of_image:
+              is_rhel: True
+              image_id: ami-123456789
+              other_key: other_value
+        name_of_profile2:
+          images:
+            name_of_image:
+              is_rhel: False
+              image_id: ami-567890234
+              other_key: other_value
+    """
+    client = aws_session(aws_profile).client('ec2')
+    cfg = config.get_aws_config()
+    image_id = cfg['profiles'][aws_profile]['images'][image_name]['image_id']
+    return client.run_instances(
+        MaxCount=count,
+        MinCount=count,
+        ImageId=image_id)
+
+
+def run_instances(aws_profile, image_name, count, run_time):
+    """Create instances and run them for a certain amount of time.
+
+    :param aws_profile: (string) Name of profile as defined in config file
+    :param image_name: (string) Name of image as defined in config file
+    :param count: (int) Number of instances to create.
+    :param run_time: (double) Desired time to let instances run (in seconds)
+
+    :returns: (double) Time from when instances were all live until all
+         instances were terminated (in seconds).
+    """
+    response = create_instances(aws_profile, image_name, count=count)
```

Comment:
Let's move the `wait_until_running()` pool inside `create_instance()` itself. That will make using it a lot cleaner and, later, if we use it in other helpers, we'll keep its usage consistently correct.

Reply:
:+1: